lib/AST/ASTPrinter.cpp

```diff
@@ -4871,6 +4888,9 @@ void PrintAST::visitDictionaryExpr(DictionaryExpr *expr) {
 }
 
 void PrintAST::visitArrowExpr(ArrowExpr *expr) {
+  visit(expr->getArgsExpr());
+  Printer << " -> ";
+  visit(expr->getResultExpr());
```

Comment:
For completeness, this should also print `async` and/or `throw` if `getAsyncLoc()` and/or `getThrowsLoc()` are valid.